src/iiiftilesource.js

```diff
@@ -333,7 +333,45 @@ $.extend( $.IIIFTileSource.prototype, $.TileSource.prototype, /** @lends OpenSea
             }
         }
 
-        return $.TileSource.prototype.getNumTiles.call(this, level);
+        // Use supplied list of scaled resolution sizes if these exist
+        var levelSize = this.getLevelSize(level);
+        if( levelSize ) {
+          var x = Math.ceil( levelSize.width / this.getTileWidth(level) ),
+              y = Math.ceil( levelSize.height / this.getTileHeight(level) );
+          return new $.Point( x, y );
+        }
+        // Otherwise call default TileSource->getNumTiles() function
+        else {
+          return $.TileSource.prototype.getNumTiles.call(this, level);
+        }
+    },
+
+
+    /**
+     * Determine image size at a given resolution level using the info.json "sizes" field
+     * Returns null if this information is not present
+     * @function {Number} level
+     */
+    getLevelSize: function( level ) {
+
+      var numLevels = this.maxLevel - this.minLevel;
+      // Need to take into account that the list may or may not include the full resolution size
+      if( this.sizes && ((this.sizes.length === numLevels) ||
+                         (this.sizes.length === numLevels + 1)) ) {
+        var levelWidth, levelHeight;
+        if( this.sizes.length === numLevels ) {
+          levelWidth = (level === this.sizes.length) ? this.width : this.sizes[level].width;
+          levelHeight = (level === this.sizes.length) ? this.height : this.sizes[level].height;
+        }
+        else {
+          levelWidth = this.sizes[level].width;
+          levelHeight = this.sizes[level].height;
+        }
+        return { width: levelWidth, height: levelHeight };
+      }
+      else {
+        return null;
+      }
```

Comment:
Here's a slight optimization to remove some redundant checks to simplify the function. (I haven't tested this, but it should be functionally equivalent).

 - Moves `this.sizes` check to the front so we can optimize it out of the first check
 - Creates a local var `sizeLength` so we don't need to constantly get the length of the size array
 - Removes the first `if` statement since we no longer need to check `this.sizes`, and the `numLevel` checks are redundant with the inner check.

```js
        if (!this.sizes) return null;
        
        var numLevels = this.maxLevel - this.minLevel;
        var sizeLength = this.sizes.length;
        
        var levelWidth, levelHeight;
        
        if (sizeLength === numLevels) {
            levelWidth = (level === sizeLength) ? this.width : this.sizes[level].width;
            levelHeight = (level === sizeLength) ? this.height : this.sizes[level].height;
        } else {
            levelWidth = this.sizes[level].width;
            levelHeight = this.sizes[level].height;
        }
        
        return {width: levelWidth, height: levelHeight};
```

Reply:
The reason I put everything in an if block was in case the sizes field doesn't in fact contain a list of resolution levels. In theory sizes could contain just a single size or even more sizes than there are levels. So what about something like this:

```
        if (!this.sizes) return null;
        
        var numLevels = this.maxLevel - this.minLevel;
        var sizeLength = this.sizes.length;
        
        var levelWidth, levelHeight;
        
        if (sizeLength === numLevels) {
            levelWidth = (level === sizeLength) ? this.width : this.sizes[level].width;
            levelHeight = (level === sizeLength) ? this.height : this.sizes[level].height;
        } else if ( sizeLength === numLevels +1 ) {
            levelWidth = this.sizes[level].width;
            levelHeight = this.sizes[level].height;
        } else {
           return null;
        }
        
        return {width: levelWidth, height: levelHeight};
```

Reply:
Works for me. @ahankinson What do you think? Thank you for reviewing :)

Reply:
I've just noticed a strange flickering effect with this code. However, I haven't been able to track down the origin. This example is using the code from the pull request: https://merovingio.c2rmf.cnrs.fr/iipimage/openseadragon/odd.html

If you shrink your browser window right down so that you can zoom out to an image only maybe 100px in width, then use the mouse scroll to very slowly zoom in, you'll see it flicker at some point, where a column of tiles is sporadically added to the right of the image. I'm guessing it's at the point when it transitions between levels, but I haven't been able to pin-point in the OSD code where this is coming from or why. @iangilman, do you think you could take a look? If I comment out the getNumTiles() function, there's no flickering, but OSD requests non-existent tiles. Maybe there's another function in addition to getNumTiles() which needs to be overridden?

Reply:
OK, after a bit of hacking, I managed to figure out and fix the flickering problem by fully implementing getTileAtPoint() when we have a list of resolutions: https://github.com/openseadragon/openseadragon/pull/2337/commits/5fd125dc92cd95461f2e033af64b19f6a620a681. By default getTileAtPoint() uses a scaling factor to calculate resolution sizes, so I needed to force it to use the sizes in our list.

Reply:
Looks good to me!